portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -39,16 +44,18 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
 
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-      int id = this.comments.size();
+      //TODO: use GSON deserialization
       String name = request.getParameter("name");
       String content = request.getParameter("content");
-      Comment comment = new Comment(id, name, content);
-      this.comments.add(comment);
+      Comment comment = new Comment(0, name, content); //Since the ID will only be assigned by the Datastore, I set it to 0. 
+      //What would be a better practice?
+      datastoreInterface.addComment(comment);
       response.sendRedirect("/index.html#commentSection");
   }
 
   private String commentsToJson(){
-    String json = gson.toJson(this.comments);
+    ArrayList<Comment> comments = datastoreInterface.getComments();
```

Comment:
In the sense of polymorphism, use the interface `List<Comment>` on the right side of assignments. Consumers shouldn't have to care about what implementation is used and just do their operations based on the available interface.

Reply:
You mean left side? `List<Comment> comments = datastoreInterface.getComments();` ?

Reply:
I think I'm not completely sure why this would be advantageous. Maybe at a later point I will have to order the comments in some way, or maybe just return some subset of them.  This probably won't happen in this function, but that is why the getComments() returns an ArrayList instead of just a List. Why should I not use the specific type it returns?

Reply:
Okay, I can see that it is possible that later I would change the return to some other kind of List, and then I wouldn't need to rewrite this function.

Reply:
Exactly!